components/utilities/chartLayout.js

```diff
@@ -149,9 +149,9 @@
             chartElements.chartArea = chart;
 
             // Defs - for clipping path
-            var defs = chart.selectAll('defs').data(dummyData);
-            defs.enter().append('defs');
-            defs.exit().remove();
+            chartElements.defs = chart.selectAll('defs').data(dummyData);
+            chartElements.defs.enter().append('defs');
+            chartElements.defs.exit().remove();
```

Comment:
Just re-read the page on data-joins, and yes, I was wrong, my suggestion wouldn't work!

I think I do prefer your proposed simplification. I find the chartLayout code a little hard to read due to this repetition.

When faced with repitition, it's also worth thinking about the use of local functions. I was able to remove some repetition here as a result:

https://github.com/ScottLogic/d3-financial-components/pull/188/files#diff-0f9b6998675c281bf770ec5c14c81ea1R195